addons/sale_timesheet/wizard/sale_make_invoice_advance.py

```diff
@@ -38,6 +38,8 @@ def _create_invoices(self, sale_orders):
             qty_to_invoice for each product_id in sale.order.line,
             before creating the invoice.
         """
+        self.sale_order_ids.tasks_ids.timesheet_ids._stop_all_users_timer()
```

Comment:
maybe it's not possible functionally, but could the timezone be validated already? in that case we would get an error:

https://github.com/odoo/enterprise/blob/348e5d0c85771f77562b9afbe75e874fea46ad98/timesheet_grid/models/analytic.py#L576-L577

if it was possible, we should filter out the timesheet_ids that are validated